src/docfx/lib/github/GitHubUserCache.cs

```diff
@@ -150,6 +154,19 @@ public async Task<Error> SaveChanges(Config config)
                 return null;
             }
 
+            var remainingRetries = 30;
+            var (error, response) = await SaveChangesCore(config, _etag);
+            while (error == null && remainingRetries-- > 0 && response.StatusCode == HttpStatusCode.PreconditionFailed)
+            {
+                var (restorePath, etag) = await RestoreFile.Restore(_url, config);
```

Comment:
Instead of changing restore, can this line call `Http.Get` directly and do the merge in memory?